models/auth/oauth2.go

```diff
@@ -394,10 +404,14 @@ func (grant *OAuth2Grant) GenerateNewAuthorizationCode(redirectURI, codeChalleng
 }
 
 func (grant *OAuth2Grant) generateNewAuthorizationCode(e db.Engine, redirectURI, codeChallenge, codeChallengeMethod string) (code *OAuth2AuthorizationCode, err error) {
-	var codeSecret string
-	if codeSecret, err = secret.New(); err != nil {
+	rBytes, err := util.CryptoRandomBytes(34)
+	if err != nil {
 		return &OAuth2AuthorizationCode{}, err
 	}
+	// Add a prefix to the base32, this is in order to make it easier
+	// for code scanners to grab sensitive tokens.
+	codeSecret := "gta_" + base32Lower.EncodeToString(rBytes)
```

Comment:
gto_ or gta_ :D

Reply:
is there a standard/rfc ?!?

Reply:
Free style ... https://datatracker.ietf.org/doc/html/rfc6749#section-1.4

```
Access tokens can have different formats, structures, and methods of
utilization (e.g., cryptographic properties) based on the resource
server security requirements. 
```

And https://datatracker.ietf.org/doc/html/rfc6749#appendix-A.2

```
     VSCHAR     = %x20-7E
     access-token = 1*VSCHAR
     client-secret = *VSCHAR
```


However, 34 random bytes is too long. AES-128 is safe enough and it only uses 16 bytes keys. I prefer to shorten our keys/tokens to 16 random bytes

Reply:
> However, 34 random bytes is still very long. AES-128 is safe enough and it only uses 16 bytes keys. I prefer to shorten our keys/tokens to 16 random bytes

I'm sorry, but I don't understand where the AES is coming from? These are strings that should be near-impossible to guess or predict what they are. These things shouldn't be used as any form of encryption key, as these are exposed to the end-user(only once). The current strings are using 44 bytes, this PR actually already reduces them without compromising security of any sort.

Reply:
As for the prefix, I might hold off onto that, as that might need a better proposal/agreement on how to use that(read never in my lifetime will happen without serious motivation).

Reply:
>  but I don't understand where the AES is coming from.

I mean, AES-128 use 16 bytes key and is considered to be secure enough. 16 bytes is already a reasonable length for secure usage, and in many cases, AES keys are generated randomly. Although we use random bytes for random ID, it's similar. It's not wrong to use longer bytes but it is not necessary to be that long for our case.

> for the prefix

I vote for the prefixes to make the strings easier to be identified.